lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -520,6 +528,9 @@ def verify_file(self, path):
     def parse(self, inventory, loader, path, cache=True):
         super(InventoryModule, self).parse(inventory, loader, path)
 
+        if get_option('use_legacy_script_group_name_sanitization'):
```

Comment:
Along with changing to self.get_option, this should move below _read_config_data (which is adding the options in the config file with set_options). Also need to import re, then this works for me.